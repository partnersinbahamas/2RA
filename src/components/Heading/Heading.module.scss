@import '../utils/styles/styles.scss';

.h1,
.h2,
.h3,
.h4,
.h5,
.h6 {
  display: block;
  font-weight: 700;
  font-family: 'Nunito Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif;
  margin-bottom: 1rem;
  margin-top: 0;
}
