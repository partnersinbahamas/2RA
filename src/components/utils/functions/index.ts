export const wait = (ms: number) =>
  new Promise(resolve => setTimeout(resolve, ms));

export function isDislayCountFormatValid(format: string) {
  const regex = /^X{1,16}$/;
  return regex.test(format);
}

export function numberFixTo(number: number | string, format: string): string {
  if (format && !isDislayCountFormatValid(format)) {
    throw new Error(
      'fixTo: fromat invalid. Please enter the correct format: X*16. Example: XXX.',
    );
  }

  return Number(number).toFixed(format?.length);
}
