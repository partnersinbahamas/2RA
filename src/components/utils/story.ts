// NOTE: story controls-options object;
export const story = {
  // controles
  name: { control: 'text' },
  abbreviation: { control: 'text' },
  badged: { control: 'boolean' },
  disabled: { control: 'boolean' },
  backgroundColor: { control: 'color' },
  src: { control: 'text' },

  // options
  stile: {
    control: 'inline-radio',
    options: ['primary', 'default', 'mute'],
  },
  size: {
    control: 'inline-radio',
    options: ['small', 'medium', 'large'],
  },
  status: {
    control: 'inline-radio',
    options: ['online', 'default', 'not-here', 'busy', 'sleep', 'primary'],
  },

  //handles
  onClick: { control: 'changed' },
};
