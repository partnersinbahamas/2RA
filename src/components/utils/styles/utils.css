.m-0 {
  margin: 0 !important;
}

/* top */
.mt-auto {
  margin-top: auto !important;
}

.mt-0 {
  margin-top: 0 !important;
}

.mt-1-2 {
  margin-top: 0.5rem !important;
}

.mt-1-4 {
  margin-top: 0.25rem !important;
}

.mt-1 {
  margin-top: 1rem !important;
}

.mt-2 {
  margin-top: 2rem !important;
}

/* right */
.mr-auto {
  margin-right: auto !important;
}

.mr-0 {
  margin-right: 0 !important;
}

.mr-1-2 {
  margin-right: 0.5rem !important;
}

.mr-1-4 {
  margin-right: 0.25rem !important;
}

.mr-1 {
  margin-right: 1rem !important;
}

.mr-2 {
  margin-right: 2rem !important;
}

/* bottom */
.mb-auto {
  margin-bottom: auto !important;
}

.mb-0 {
  margin-bottom: 0 !important;
}

.mb-1-2 {
  margin-bottom: 0.5rem !important;
}

.mb-1-4 {
  margin-bottom: 0.25rem !important;
}

.mb-1 {
  margin-bottom: 1rem !important;
}

.mb-2 {
  margin-bottom: 2rem !important;
}

/* left */
.ml-auto {
  margin-left: auto !important;
}

.ml-0 {
  margin-left: 0 !important;
}

.ml-1-2 {
  margin-left: 0.5rem !important;
}

.ml-1-4 {
  margin-left: 0.25rem !important;
}

.ml-1 {
  margin-left: 1rem !important;
}

.ml-2 {
  margin-left: 2rem !important;
}
