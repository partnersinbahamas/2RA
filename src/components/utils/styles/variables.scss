$breakpoint-mobile: 576px; // minimal width for mobile
$breakpoint-tablet: 768px; // minimal width for tablet
$breakpoint-desktop: 1200px; // minimal width for desktop
