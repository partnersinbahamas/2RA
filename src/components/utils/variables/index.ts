import { TStile } from '../types/types';
export const stiles: TStile[] = ['mute', 'default', 'primary'];
export const levels: TLevel[] = [1, 2, 3, 4, 5, 6];
