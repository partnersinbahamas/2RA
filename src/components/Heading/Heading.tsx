import classNames from 'classnames';
import PropsTypes from 'prop-types';
import styles from './Heading.module.scss';

type TProps = React.ComponentProps<'hgroup'> & {
  level: TLevel;
};

const Heading: React.FC<TProps> = ({
  level,
  children = `Heading ${level}`,
  ...props
}) => {
  const { className, ...headingProps } = props;

  const tag = 'h';
  const tagName = `${tag}${level}`;
  const Tag = tagName as React.ElementType;

  return (
    <Tag
      {...headingProps}
      role="heading"
      aria-level={level}
      className={classNames(styles[`${tagName}`], className)}
    >
      {children}
    </Tag>
  );
};

export default Heading;

Heading.propTypes = {
  level: PropsTypes.any,
  className: PropsTypes.string,
  children: PropsTypes.node.isRequired,
};
