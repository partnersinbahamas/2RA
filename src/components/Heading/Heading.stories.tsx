import { Meta, StoryObj } from '@storybook/react';
import Heading from './Heading';
import { story } from '../utils/variables/story';

const meta = {
  title: 'Athomic/Heading',
  component: Heading,
  tags: ['autodocs'],
  args: { level: 1 },
  argTypes: {
    level: { control: { type: 'number', min: 1, max: 6 } },
    children: story.children,
  },
} satisfies Meta<typeof Heading>;
export default meta;

type TStory = StoryObj<typeof meta>;

export const h1: TStory = { args: { level: 1 } };
export const h2: TStory = { args: { level: 2 } };
export const h3: TStory = { args: { level: 3 } };
export const h4: TStory = { args: { level: 4 } };
export const h5: TStory = { args: { level: 5 } };
export const h6: TStory = { args: { level: 6 } };
