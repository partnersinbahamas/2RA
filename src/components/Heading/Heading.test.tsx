import { render, screen } from '@testing-library/react';
import Heading from './Heading';
import { levels } from '../utils/variables';

describe('Heading', () => {
  it.each(levels)('renders correctly', level => {
    const child = `Heading level ${level}`;
    const { container } = render(<Heading level={level}>{child}</Heading>);

    const headingElement = screen.getByRole('heading');
    expect(headingElement.tagName).toBe(`H${level}`);
    expect(headingElement).toHaveTextContent(child);

    expect(container).toMatchSnapshot();
  });
});
