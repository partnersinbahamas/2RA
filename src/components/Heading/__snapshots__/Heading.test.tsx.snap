// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders correctly 1`] = `
<div>
  <h1
    aria-level="1"
    class="h1"
    role="heading"
  >
    Heading level 1
  </h1>
</div>
`;

exports[`Heading renders correctly 2`] = `
<div>
  <h2
    aria-level="2"
    class="h2"
    role="heading"
  >
    Heading level 2
  </h2>
</div>
`;

exports[`Heading renders correctly 3`] = `
<div>
  <h3
    aria-level="3"
    class="h3"
    role="heading"
  >
    Heading level 3
  </h3>
</div>
`;

exports[`Heading renders correctly 4`] = `
<div>
  <h4
    aria-level="4"
    class="h4"
    role="heading"
  >
    Heading level 4
  </h4>
</div>
`;

exports[`Heading renders correctly 5`] = `
<div>
  <h5
    aria-level="5"
    class="h5"
    role="heading"
  >
    Heading level 5
  </h5>
</div>
`;

exports[`Heading renders correctly 6`] = `
<div>
  <h6
    aria-level="6"
    class="h6"
    role="heading"
  >
    Heading level 6
  </h6>
</div>
`;
